cmd/syncthing/gui.go

```diff
@@ -1567,15 +1567,51 @@ func browseFiles(current string, fsType fs.FilesystemType) []string {
 
 	fs := fs.NewFilesystem(fsType, searchDir)
 
-	subdirectories, _ := fs.Glob(searchFile + "*")
+	subdirectories, _ := fs.DirNames("")
 
-	ret := make([]string, 0, len(subdirectories))
+	type match struct {
+		path             string
+		case_differences int
+	}
```

Comment:
Don't make inline structs in functions. Put it at file scope. Also, no snake case in go.